utilities/memory_allocators.h

```diff
@@ -0,0 +1,65 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <atomic>
+
+#include "rocksdb/memory_allocator.h"
+
+namespace ROCKSDB_NAMESPACE {
+// A memory allocator using new/delete
```

Comment:
Need some comments to explain when should the user use this default memory allocator. Currently, it is mainly used in the test to replace the "CustomMemoryAllocator"

Reply:
There is no great advantage of using this class.  We could set it as the default allocator and get rid of all of the code that checks if one exists and does new if it does not.

Reply:
I see. Just add some comments here to indicate the potential use cases.